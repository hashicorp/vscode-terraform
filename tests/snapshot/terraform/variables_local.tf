locals {
  service_name = "forum"
  owner        = "Community Team"
}

locals {
  # Ids for multiple sets of EC2 instances, merged together
  instance_ids = concat(aws_instance.blue.*.id, aws_instance.green.*.id)
}

locals {
  # Common tags to be assigned to all resources
  common_tags = {
    Service = local.service_name
    Owner   = local.owner
  }
}

resource "aws_instance" "example" {
  # ...

  tags = local.common_tags
}
