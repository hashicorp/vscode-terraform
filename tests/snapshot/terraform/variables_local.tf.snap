>locals {
#^^^^^^ source.terraform meta.block.terraform entity.name.type.terraform
#      ^ source.terraform meta.block.terraform
#       ^ source.terraform meta.block.terraform punctuation.section.block.begin.terraform
>  service_name = "forum"
#^^ source.terraform meta.block.terraform
#  ^^^^^^^^^^^^ source.terraform meta.block.terraform variable.declaration.terraform variable.other.readwrite.terraform
#              ^ source.terraform meta.block.terraform variable.declaration.terraform
#               ^ source.terraform meta.block.terraform variable.declaration.terraform keyword.operator.assignment.terraform
#                ^ source.terraform meta.block.terraform variable.declaration.terraform
#                 ^ source.terraform meta.block.terraform string.quoted.double.terraform punctuation.definition.string.begin.terraform
#                  ^^^^^ source.terraform meta.block.terraform string.quoted.double.terraform
#                       ^ source.terraform meta.block.terraform string.quoted.double.terraform punctuation.definition.string.end.terraform
>  owner        = "Community Team"
#^^ source.terraform meta.block.terraform
#  ^^^^^ source.terraform meta.block.terraform variable.declaration.terraform variable.other.readwrite.terraform
#       ^^^^^^^^ source.terraform meta.block.terraform variable.declaration.terraform
#               ^ source.terraform meta.block.terraform variable.declaration.terraform keyword.operator.assignment.terraform
#                ^ source.terraform meta.block.terraform variable.declaration.terraform
#                 ^ source.terraform meta.block.terraform string.quoted.double.terraform punctuation.definition.string.begin.terraform
#                  ^^^^^^^^^^^^^^ source.terraform meta.block.terraform string.quoted.double.terraform
#                                ^ source.terraform meta.block.terraform string.quoted.double.terraform punctuation.definition.string.end.terraform
>}
#^ source.terraform meta.block.terraform punctuation.section.block.end.terraform
>
>locals {
#^^^^^^ source.terraform meta.block.terraform entity.name.type.terraform
#      ^ source.terraform meta.block.terraform
#       ^ source.terraform meta.block.terraform punctuation.section.block.begin.terraform
>  # Ids for multiple sets of EC2 instances, merged together
#^^ source.terraform meta.block.terraform
#  ^ source.terraform meta.block.terraform comment.line.number-sign.hcl punctuation.definition.comment.terraform
#   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.terraform meta.block.terraform comment.line.number-sign.hcl
>  instance_ids = concat(aws_instance.blue.*.id, aws_instance.green.*.id)
#^^ source.terraform meta.block.terraform
#  ^^^^^^^^^^^^ source.terraform meta.block.terraform variable.declaration.terraform variable.other.readwrite.terraform
#              ^ source.terraform meta.block.terraform variable.declaration.terraform
#               ^ source.terraform meta.block.terraform variable.declaration.terraform keyword.operator.assignment.terraform
#                ^ source.terraform meta.block.terraform variable.declaration.terraform
#                 ^^^^^^ source.terraform meta.block.terraform meta.function-call.terraform support.function.builtin.terraform
#                       ^ source.terraform meta.block.terraform meta.function-call.terraform punctuation.section.parens.begin.terraform
#                        ^^^^^^^^^^^^ source.terraform meta.block.terraform meta.function-call.terraform
#                                    ^ source.terraform meta.block.terraform meta.function-call.terraform keyword.operator.accessor.terraform
#                                     ^^^^ source.terraform meta.block.terraform meta.function-call.terraform variable.other.member.terraform
#                                         ^ source.terraform meta.block.terraform meta.function-call.terraform keyword.operator.accessor.terraform
#                                          ^^ source.terraform meta.block.terraform meta.function-call.terraform
#                                            ^^ source.terraform meta.block.terraform meta.function-call.terraform variable.other.member.terraform
#                                              ^ source.terraform meta.block.terraform meta.function-call.terraform punctuation.separator.terraform
#                                               ^^^^^^^^^^^^^ source.terraform meta.block.terraform meta.function-call.terraform
#                                                            ^ source.terraform meta.block.terraform meta.function-call.terraform keyword.operator.accessor.terraform
#                                                             ^^^^^ source.terraform meta.block.terraform meta.function-call.terraform variable.other.member.terraform
#                                                                  ^ source.terraform meta.block.terraform meta.function-call.terraform keyword.operator.accessor.terraform
#                                                                   ^^ source.terraform meta.block.terraform meta.function-call.terraform
#                                                                     ^^ source.terraform meta.block.terraform meta.function-call.terraform variable.other.member.terraform
#                                                                       ^ source.terraform meta.block.terraform meta.function-call.terraform punctuation.section.parens.end.terraform
>}
#^ source.terraform meta.block.terraform punctuation.section.block.end.terraform
>
>locals {
#^^^^^^ source.terraform meta.block.terraform entity.name.type.terraform
#      ^ source.terraform meta.block.terraform
#       ^ source.terraform meta.block.terraform punctuation.section.block.begin.terraform
>  # Common tags to be assigned to all resources
#^^ source.terraform meta.block.terraform
#  ^ source.terraform meta.block.terraform comment.line.number-sign.hcl punctuation.definition.comment.terraform
#   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.terraform meta.block.terraform comment.line.number-sign.hcl
>  common_tags = {
#^^ source.terraform meta.block.terraform
#  ^^^^^^^^^^^ source.terraform meta.block.terraform variable.declaration.terraform variable.other.readwrite.terraform
#             ^ source.terraform meta.block.terraform variable.declaration.terraform
#              ^ source.terraform meta.block.terraform variable.declaration.terraform keyword.operator.assignment.terraform
#               ^ source.terraform meta.block.terraform variable.declaration.terraform
#                ^ source.terraform meta.block.terraform meta.braces.terraform punctuation.section.braces.begin.terraform
>    Service = local.service_name
#^^^^ source.terraform meta.block.terraform meta.braces.terraform
#    ^^^^^^^ source.terraform meta.block.terraform meta.braces.terraform meta.mapping.key.terraform variable.other.readwrite.terraform
#           ^ source.terraform meta.block.terraform meta.braces.terraform
#            ^ source.terraform meta.block.terraform meta.braces.terraform keyword.operator.assignment.terraform
#             ^ source.terraform meta.block.terraform meta.braces.terraform
#              ^^^^^ source.terraform meta.block.terraform meta.braces.terraform support.constant.terraform
#                   ^^^^^^^^^^^^^^ source.terraform meta.block.terraform meta.braces.terraform
>    Owner   = local.owner
#^^^^ source.terraform meta.block.terraform meta.braces.terraform
#    ^^^^^ source.terraform meta.block.terraform meta.braces.terraform meta.mapping.key.terraform variable.other.readwrite.terraform
#         ^^^ source.terraform meta.block.terraform meta.braces.terraform
#            ^ source.terraform meta.block.terraform meta.braces.terraform keyword.operator.assignment.terraform
#             ^ source.terraform meta.block.terraform meta.braces.terraform
#              ^^^^^ source.terraform meta.block.terraform meta.braces.terraform support.constant.terraform
#                   ^^^^^^^ source.terraform meta.block.terraform meta.braces.terraform
>  }
#^^ source.terraform meta.block.terraform meta.braces.terraform
#  ^ source.terraform meta.block.terraform meta.braces.terraform punctuation.section.braces.end.terraform
>}
#^ source.terraform meta.block.terraform punctuation.section.block.end.terraform
>
>resource "aws_instance" "example" {
#^^^^^^^^ source.terraform meta.block.terraform entity.name.type.terraform
#        ^ source.terraform meta.block.terraform
#         ^ source.terraform meta.block.terraform entity.name.tag.terraform entity.name.tag.begin.terraform
#          ^^^^^^^^^^^^ source.terraform meta.block.terraform entity.name.tag.terraform
#                      ^ source.terraform meta.block.terraform entity.name.tag.terraform entity.name.tag.end.terraform
#                       ^ source.terraform meta.block.terraform
#                        ^ source.terraform meta.block.terraform entity.name.tag.terraform entity.name.tag.begin.terraform
#                         ^^^^^^^ source.terraform meta.block.terraform entity.name.tag.terraform
#                                ^ source.terraform meta.block.terraform entity.name.tag.terraform entity.name.tag.end.terraform
#                                 ^ source.terraform meta.block.terraform
#                                  ^ source.terraform meta.block.terraform punctuation.section.block.begin.terraform
>  # ...
#^^ source.terraform meta.block.terraform
#  ^ source.terraform meta.block.terraform comment.line.number-sign.hcl punctuation.definition.comment.terraform
#   ^^^^ source.terraform meta.block.terraform comment.line.number-sign.hcl
>
>  tags = local.common_tags
#^^ source.terraform meta.block.terraform
#  ^^^^ source.terraform meta.block.terraform variable.declaration.terraform variable.other.readwrite.terraform
#      ^ source.terraform meta.block.terraform variable.declaration.terraform
#       ^ source.terraform meta.block.terraform variable.declaration.terraform keyword.operator.assignment.terraform
#        ^ source.terraform meta.block.terraform variable.declaration.terraform
#         ^^^^^ source.terraform meta.block.terraform support.constant.terraform
#              ^ source.terraform meta.block.terraform keyword.operator.accessor.terraform
#               ^^^^^^^^^^^ source.terraform meta.block.terraform variable.other.member.terraform
>}
#^ source.terraform meta.block.terraform punctuation.section.block.end.terraform
>